src/groups/mqb/mqbnet/mqbnet_tcpsessionfactory.cpp

```diff
@@ -680,6 +689,12 @@ void TCPSessionFactory::channelStateCallback(
             channel->close(closeStatus);
         }
         else {
+            {  // Save begin session timestamp
+                bslmt::LockGuard<bslmt::Mutex> guard(&d_mutex);  // LOCK
+                d_timestampMap[channel.get()] =
+                    mwcsys::Time::highResolutionTimer();
```

Comment:
I am concerned about these synchronizations between different threads on channel operations. I believe it's possible to store these timestamps directly in one of the `mwcio::Channel` implementations, so we don't need a mutex synchronization for them at all.

Let's add TODO for this

Reply:
Agree, but for this it's needed to change interface and implementations. So added TODO for now.